cabal-install/tests/IntegrationTests2.hs

```diff
@@ -93,7 +95,16 @@ removePathForcibly = removeDirectoryRecursive
 #endif
 
 main :: IO ()
-main =
+main = do
+  -- this is needed to ensure tests aren't affected by the user's cabal config
```

Comment:
Would be good to have a GitHub issue with more explanation and examples, and link it both from the PR and from here.

Reply:
I linked the issue I'd opened for this; let me know what else needs to be documented there.